diff64.c

```diff
@@ -77,12 +77,37 @@ static uint32_t get_bit(const uint64_t number, const uint32_t index)
     return ((number >> index) & 1uLL);
 }
 
+static uint32_t get_max_number_length(const uint64_t left_number, const uint64_t right_number)
+{
+    uint32_t left_number_length;
+    uint32_t right_number_length;
+    uint32_t max_number_length;
+    const uint32_t max_string_length = MAX_COMPARE_LENGTH/4 + 1;
+    char number_length[max_string_length];
+
+    snprintf(number_length, max_string_length, "%lX", left_number);
+    left_number_length = strlen(number_length);
+    snprintf(number_length, max_string_length, "%lX", right_number);
```

Comment:
I prefer lower case "x", not that it really matters here but for the sake of consistency, if nothing else =)